### PR TITLE
Changed all 'NED board' references to 'FRD body'. Also cleaned up mix…

### PR DESCRIPTION
**Pull request by rblilja**
…ing of m/s/s and m/s^2 to use the latter. Corrected m/s/s to Pascals. Plus minor typos. There are still some messages that needs to be looked at for correct reference frame: 'vehicle_rates_setpoint.msg' and 'vehicle_attitude_setpoint.msg. Also made some minor cosmetic clean ups.

Solves: #16328

Note: This PR replaces PR #16337

### CODE REVIEW

**Comment by rblilja** [comment]
@dagar @bresch New PR. Closed previous one (#16337) as it got messy. My bad.

File: msg/sensor_combined.msg [hunk 1]
```diff
@@ -7,12 +7,12 @@ uint64 timestamp				# time since system start (microseconds)
...
 # gyro timstamp is equal to the timestamp of the message
-float32[3] gyro_rad			# average angular rate measured in the XYZ body frame in rad/s over the last gyro sampling period
-uint32 gyro_integral_dt		# gyro measurement sampling period in us
+float32[3] gyro_rad			# average angular rate measured in the FRD body frame XYZ-axis in rad/s over the last gyro sampling period
+uint32 gyro_integral_dt		# gyro measurement sampling period in microseconds
 
 int32 accelerometer_timestamp_relative	# timestamp + accelerometer_timestamp_relative = Accelerometer timestamp
-float32[3] accelerometer_m_s2		# average value acceleration measured in the XYZ body frame in m/s/s over the last accelerometer sampling period
-uint32 accelerometer_integral_dt	# accelerometer measurement sampling period in us
+float32[3] accelerometer_m_s2		# average value acceleration measured in the FRD body frame XYZ-axis in m/s/s over the last accelerometer sampling period
```
Comment (bresch) [inline msg/sensor_combined.msg:14]:
```suggestion
float32[3] accelerometer_m_s2		# average value acceleration measured in the FRD body frame XYZ-axis in m/s^2 over the last accelerometer sampling period
```

File: msg/sensors_status_imu.msg [hunk 2]
```diff
@@ -6,7 +6,7 @@ uint64 timestamp # time since system start (microseconds)
 uint32 accel_device_id_primary       # current primary accel device id for reference
 
 uint32[4] accel_device_ids
-float32[4] accel_inconsistency_m_s_s # magnitude of acceleration difference between IMU instance and mean in (m/s/s).
+float32[4] accel_inconsistency_m_s_s # magnitude of acceleration difference between IMU instance and mean in (m/s^2).
```
Comment (bresch) [inline msg/sensors_status_imu.msg:9]:
```suggestion
float32[4] accel_inconsistency_m_s_s # magnitude of acceleration difference between IMU instance and mean in m/s^2.
```

**Review by bresch — CHANGES_REQUESTED** [review]
Thanks for the PR; just 2 small things, the rest seems good

**Comment by rblilja** [comment]
Good catch! Shall be included now.

File: msg/sensor_correction.msg [hunk 3]
```diff
@@ -15,15 +15,15 @@ float32[3] gyro_offset_3	# gyro 3 XYZ offsets in the sensor frame in rad/s
...
 # Note the corrections are in the sensor frame and must be applied before the sensor data is rotated into body frame
 uint32[4] accel_device_ids
-float32[3] accel_offset_0	# accelerometer 0 XYZ offsets in the sensor frame in m/s/s
-float32[3] accel_offset_1	# accelerometer 1 XYZ offsets in the sensor frame in m/s/s
-float32[3] accel_offset_2	# accelerometer 2 XYZ offsets in the sensor frame in m/s/s
-float32[3] accel_offset_3	# accelerometer 3 XYZ offsets in the sensor frame in m/s/s
+float32[3] accel_offset_0	# accelerometer 0 XYZ offsets in the sensor frame in m/s^s
+float32[3] accel_offset_1	# accelerometer 1 XYZ offsets in the sensor frame in m/s^s
+float32[3] accel_offset_2	# accelerometer 2 XYZ offsets in the sensor frame in m/s^s
+float32[3] accel_offset_3	# accelerometer 3 XYZ offsets in the sensor frame in m/s^s
```
Comment (bresch) [inline msg/sensor_correction.msg:21]:
```suggestion
float32[3] accel_offset_0	# accelerometer 0 offsets in the FRD board frame XYZ-axis in m/s^s
float32[3] accel_offset_1	# accelerometer 1 offsets in the FRD board frame XYZ-axis in m/s^s
float32[3] accel_offset_2	# accelerometer 2 offsets in the FRD board frame XYZ-axis in m/s^s
float32[3] accel_offset_3	# accelerometer 3 offsets in the FRD board frame XYZ-axis in m/s^s
```

File: msg/sensor_gyro.msg [hunk 4]
```diff
@@ -3,11 +3,11 @@ uint64 timestamp_sample
 
 uint32 device_id          # unique device ID for the sensor that does not change between power cycles
 
-float32 x                 # angular velocity in the NED X board axis in rad/s
-float32 y                 # angular velocity in the NED Y board axis in rad/s
-float32 z                 # angular velocity in the NED Z board axis in rad/s
+float32 x                 # angular velocity in the FRD body frame X-axis in rad/s
+float32 y                 # angular velocity in the FRD body frame Y-axis in rad/s
+float32 z                 # angular velocity in the FRD body frame Z-axis in rad/s
```
Comment (bresch) [inline msg/sensor_gyro.msg:8]:
```suggestion
float32 x                 # angular velocity in the FRD board frame X-axis in rad/s
float32 y                 # angular velocity in the FRD board frame Y-axis in rad/s
float32 z                 # angular velocity in the FRD board frame Z-axis in rad/s
```

File: msg/sensor_gyro_fifo.msg [hunk 5]
```diff
@@ -8,8 +8,8 @@ float32 scale
 
 uint8 samples             # number of valid samples
 
-int16[32] x               # angular velocity in the NED X board axis in rad/s
-int16[32] y               # angular velocity in the NED Y board axis in rad/s
-int16[32] z               # angular velocity in the NED Z board axis in rad/s
+int16[32] x               # angular velocity in the FRD body frame X-axis in rad/s
+int16[32] y               # angular velocity in the FRD body frame Y-axis in rad/s
+int16[32] z               # angular velocity in the FRD body frame Z-axis in rad/s
```
Comment (bresch) [inline msg/sensor_gyro_fifo.msg:13]:
```suggestion
int16[32] x               # angular velocity in the FRD board frame X-axis in rad/s
int16[32] y               # angular velocity in the FRD board frame Y-axis in rad/s
int16[32] z               # angular velocity in the FRD board frame Z-axis in rad/s
```

File: msg/sensor_mag.msg [hunk 6]
```diff
@@ -3,11 +3,11 @@ uint64 timestamp_sample
 
 uint32 device_id          # unique device ID for the sensor that does not change between power cycles
 
-float32 x                 # magnetic field in the NED X board axis in Gauss
-float32 y                 # magnetic field in the NED Y board axis in Gauss
-float32 z                 # magnetic field in the NED Z board axis in Gauss
+float32 x                 # magnetic field in the FRD body frame X-axis in Gauss
+float32 y                 # magnetic field in the FRD body frame Y-axis in Gauss
+float32 z                 # magnetic field in the FRD body frame Z-axis in Gauss
```
Comment (bresch) [inline msg/sensor_mag.msg:8]:
```suggestion
float32 x                 # magnetic field in the FRD board frame X-axis in Gauss
float32 y                 # magnetic field in the FRD board frame Y-axis in Gauss
float32 z                 # magnetic field in the FRD board frame Z-axis in Gauss
```

**Review by bresch — CHANGES_REQUESTED** [review]
I went through it in details and there was still a bit of confusion (due to my comment in the issue).

To make it clear:
- all the `sensor_*` are in the "autopilot hardware frame" (called "board"). This is not necessarily the frame of the MEMS itself.
- all the `vehicle_*` are in the vehicle body frame

For example:
The data of an accelerometer is sampled, rotated in the board frame (i.e.:autopilot hardware, e.g.: Pixhawk) and published as `sensor_accel`. Then the calibrations and estimated bias are applied and the result is rotated in body frame and re-published as `vehicle_imu` and potentially in `sensors_combined` (if it has the highest priority).

**Review by bresch — APPROVED** [review]

**Comment by rblilja** [comment]
Makes sense. Good work! Perhaps it is a good idea to define and list the coordinate systems in the documentation for completeness. I have not really managed to find anything in the documentation other than the mentioning of NED and FRD systems here: https://dev.px4.io/v1.11/en/contribute/notation.html

Note that FRD in the documentation refers to the vehicle front axis etc. i.e. rather easy to misconcive mentioning of 'body' and 'board' in the RTPS message comments.

Also, to me 'board' is somewhat vague as it might refer to the complete flight controller, or just the sensor board of the Cube if you are using such flight controller. Perhaps obvious to some people, but if you have whatever diagnosis a psychiatrist would label me with, it keeps you awake at night :)

File: msg/sensor_accel_fifo.msg [hunk 7]
```diff
@@ -8,8 +8,8 @@ float32 scale
 
 uint8 samples             # number of valid samples
 
-int16[32] x               # acceleration in the NED X board axis in m/s/s
-int16[32] y               # acceleration in the NED Y board axis in m/s/s
-int16[32] z               # acceleration in the NED Z board axis in m/s/s
+int16[32] x               # acceleration in the FRD board frame X-axis in m/s^2
```
Comment (dagar) [inline msg/sensor_accel_fifo.msg:11]: To add to the confusion these FIFO messages are the raw data that we get out of an IMU's FIFO. They're in the sensor frame and the data is raw 16 bit integers, typically with UINT16_MAX corresponding to the configured max range (eg 16 G accelerometer). That's why the message carries a rotation and scale factor.

Comment (bresch) [reply msg/sensor_accel_fifo.msg:11]: Ouch, so they're not always in the same frame as `sensor_accel` ?

**Pull request merged**